lib/ansible/module_utils/network/skydive/skydive.py

```diff
@@ -0,0 +1,171 @@
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+# ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+# WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED.
+# IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+# INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+# PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+# LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE
+# USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+#
+
+from ansible.module_utils.six import iteritems
+from ansible.module_utils._text import to_text
+from ansible.module_utils.basic import env_fallback
+
+try:
+    from skydive.graph import Node, Edge
+    from skydive.rest.client import RESTClient
+    from skydive.websocket.client import WSClient
+    from skydive.websocket.client import WSClientDebugProtocol
+    from skydive.websocket.client import WSMessage
+    from skydive.websocket.client import NodeAddedMsgType, EdgeAddedMsgType
+    HAS_SKYDIVE_CLIENT = True
+except ImportError:
+    HAS_SKYDIVE_CLIENT = False
+
+# defining skydive constants
+SKYDIVE_LOOKUP_QUERY = 'G.V().Has'
+
+SKYDIVE_PROVIDER_SPEC = {
+    'host': dict(fallback=(env_fallback, ['SKYDIVE_HOST'])),
+    'username': dict(fallback=(env_fallback, ['SKYDIVE_USERNAME'])),
+    'password': dict(fallback=(env_fallback, ['SKYDIVE_PASSWORD']), no_log=True),
+    'ssl_verify': dict(type='bool', default=False, fallback=(env_fallback, ['SKYDIVE_SSL_VERIFY']))
+}
+
+
+class skydive_base(object):
+    ''' Base class for implementing Skydive Rest API '''
+    provider_spec = {'provider': dict(type='dict', options=SKYDIVE_PROVIDER_SPEC)}
+
+    def __init__(self, host="localhost:8082", username="admin", password="password"):
+        try:
+            self.restclient_object = RESTClient(host, username=username, password=password)
+            self.websocketclient_object = WSClient(host, username=username, password=password)
+        except TypeError as err:
+            raise err
+
+
+class skydive_lookup(skydive_base):
+
+    def __init__(self, provider):
+        host = provider['host']
+        username = provider['username']
+        password = provider['password']
+        super(skydive_lookup, self).__init__(host, username, password)
+        self.query_str = ""
+
+    def lookup_query(self, filter_data):
+        query_key = filter_data.keys()[0]
+        self.query_str = SKYDIVE_LOOKUP_QUERY + "('{0}', '{1}')".format(query_key, filter_data[query_key])
+        nodes = self.restclient_object.lookup_nodes(self.query_str)
+        return nodes[0].__dict__
+
+
+class skydive_flow_topology(skydive_base):
```

Comment:
Supporting captures is great, you may also want to support injecting traffic.

Reply:
Noted, shall add the support in new PR